library/std/src/thread/mod.rs

```diff
@@ -1361,6 +1361,15 @@ impl Thread {
         unsafe { self.inner.as_ref().parker().park() }
     }
 
+    /// Like the public [`park_timeout`], but callable on any handle. This is used to
+    /// allow parking in TLS destructors.
+    ///
+    /// # Safety
+    /// May only be called from the thread to which this handle belongs.
+    pub(crate) unsafe fn park_timeout(&self, dur: Duration) {
+        unsafe { self.inner.as_ref().parker().park_timeout(dur) }
```

Comment:
The public park_timeout has a PanicGuard -- should this also have that?

It looks like the ~only advantage to this is that we avoid re-fetching the handle from thread local storage, is that just a performance optimization?